### ISSUE TITLE
Cannot suppress SwallowedException on catch variable

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#4154]
## Expected Behavior
Below code passes SwallowedException check.

## Observed Behavior
```
exceptions - 20min debt
    SwallowedException - [f] at src\main\kotlin\test.kt:4:4
```

## Steps to Reproduce
```kt
fun f() {
	try {
            // more stuff
	} catch (@Suppress("SwallowedException") ex: AssertionError) {
            // more stuff
	}
}
```

## Context
Only this pattern seems to work which has a very wide effect range. Any code inside try and inside catch is also suppressed. Can't target only this specific catch.

## Your Environment

* Version of detekt used: `io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.18.1`
* Version of Gradle used (if applicable): 6.9
* Gradle scan link (add `--scan` option when running the gradle task): -
* Operating System and version: Windows 10
* Link to your project (if it's a public repository): -

**Comment by TWiStErRob** [comment]
Might be a feature request. I would class it as a "usability problem".

**Comment by chao2zhang** [comment]
Agreed. Currently, we only support suppressing in the `KtTryExpression`. So the code below would work in the short-term without waiting for the next detekt release.
```
@Suppress("SwallowedException")
try {
  // more stuff
} catch (ex: AssertionError) {
  // more stuff
}
```

**Comment by blundell** [comment]
Would be nice if variables named `ignore` where ignored:

```
} catch (ignore: AssertionError) {
```

**Comment by blundell** [comment]
annd this was from 2021, ignore me, you can configure it:

<img width="647" alt="image" src="https://github.com/detekt/detekt/assets/655860/6955b133-43cf-4a47-b841-f9f4d70110b9">

https://detekt.dev/docs/rules/exceptions/
